### PR TITLE
GameDB: Fahrenheit asylum crashing fixes.

### PR DESCRIPTION
**Pull request by atomic83GitHub**
This PR add a patch to solve a crash in the game "Fahrenheit".

Tested by atomic83github

### CODE REVIEW

File: bin/GameIndex.dbf [hunk 1]
```diff
@@ -14569,11 +14569,51 @@ Serial = SLES-53539
 Name   = Fahrenheit
 Region = PAL-M4
 Compat = 5
+[patches]
+	author=atomic83github
+	// Found with the help of the dev-9 plugin and the PS4 emu.
+	// This is a game bug trigger by a timing issue.
+	// The game have an internal crash reporting system
+	// Which rely on a debug server connected to the console.
```
Comment (pgert) [inline bin/GameIndex.dbf]: "which" not "Which".

File: bin/GameIndex.dbf [hunk 2]
```diff
@@ -14569,11 +14569,51 @@ Serial = SLES-53539
...
+	// When the game do an error, it expect the debug server to be connected
```
Comment (pgert) [inline bin/GameIndex.dbf]: "expects" not "expect".

File: bin/GameIndex.dbf [hunk 3]
```diff
@@ -14569,11 +14569,51 @@ Serial = SLES-53539
...
+	// To send the crash dump.
```
Comment (pgert) [inline bin/GameIndex.dbf]: "to" not "To".

File: bin/GameIndex.dbf [hunk 4]
```diff
@@ -14569,11 +14569,51 @@ Serial = SLES-53539
...
+	// But that never happen on PCSX2 as the debug server is not present.
+	// Thus, the game will enter in an infinte loop at the asylum level.
+	// This only happen when certain conditions are meet in the game.
+	// This patch makes the game to exit the loop by cancelling the debug reporting.
+	patch=0,EE,001c5958,word,00000000 // QDT::KCOM::COM_SERVICE::RetrieveComHandle
+	patch=0,EE,001c5b6c,word,00000000 // QDT::KCOM::COM_SERVICE::RetrieveComHandle
+	patch=0,EE,001c5d24,word,00000000 // QDT::KCOM::COM_SERVICE::RetrieveComHandle
+	// The game have no frame pacing system in place.
+	// This cause massive slowsdown in loading screens.
```
Comment (pgert) [inline bin/GameIndex.dbf]: "causes a massive slowsdown".

**Comment by mirh** [comment]
If there is a timing issue or something, there should be at least an issue to investigate it?

**Comment by atomic83GitHub** [comment]
Well, it's a complicated issue, the timing issue is caused by a failing branch that should gace the EE a few more cycle as expected by the game but it end in BC0F instead because PCSX2 don't emulate the EE pipeline bug.

**Comment by atomic83GitHub** [comment]
https://github.com/PCSX2/pcsx2/issues/2582

File: bin/GameIndex.dbf [hunk 5]
```diff
@@ -14569,11 +14569,51 @@ Serial = SLES-53539
...
+[patches]
+	author=atomic83github
+	// Found with the help of the dev-9 plugin and the PS4 emu.
+	// This is a game bug trigger by a timing issue.
+	// The game have an internal crash reporting system
+	// which rely on a debug server connected to the console.
+	// When the game do an error, it expects the debug server to be connected
+	// to send the crash dump.
+	// But that never happen on PCSX2 as the debug server is not present.
+	// Thus, the game will enter in an infinte loop at the asylum level.
```
Comment (lightningterror) [inline bin/GameIndex.dbf]: Infinite.

**Comment by lightningterror** [comment]
Will be nice if the commit description is updated, you can use the same comments from the patch itself.

File: bin/GameIndex.dbf [hunk 6]
```diff
@@ -14569,11 +14569,51 @@ Serial = SLES-53539
...
+	// to send the crash dump.
+	// But that never happen on PCSX2 as the debug server is not present.
+	// Thus, the game will enter in an infinite loop at the asylum level.
+	// This only happen when certain conditions are meet in the game.
+	// This patch makes the game to exit the loop by cancelling the debug reporting.
+	patch=0,EE,001c5958,word,00000000 // QDT::KCOM::COM_SERVICE::RetrieveComHandle
+	patch=0,EE,001c5b6c,word,00000000 // QDT::KCOM::COM_SERVICE::RetrieveComHandle
+	patch=0,EE,001c5d24,word,00000000 // QDT::KCOM::COM_SERVICE::RetrieveComHandle
+	// The game have no frame pacing system in place.
+	// This causes a massive slowsdown in loading screens.
```
Comment (lightningterror) [inline bin/GameIndex.dbf]: Slowdown.

File: bin/GameIndex.dbf [hunk 7]
```diff
@@ -14569,11 +14569,51 @@ Serial = SLES-53539
...
+	// to send the crash dump.
+	// But that never happen on PCSX2 as the debug server is not present.
+	// Thus, the game will enter in an infinte loop at the asylum level.
+	// This only happen when certain conditions are meet in the game.
+	// This patch makes the game to exit the loop by cancelling the debug reporting.
+	patch=0,EE,001c5958,word,00000000 // QDT::KCOM::COM_SERVICE::RetrieveComHandle
+	patch=0,EE,001c5b6c,word,00000000 // QDT::KCOM::COM_SERVICE::RetrieveComHandle
+	patch=0,EE,001c5d24,word,00000000 // QDT::KCOM::COM_SERVICE::RetrieveComHandle
+	// The game have no frame pacing system in place.
+	// This causes a massive slowsdown in loading screens.
```
Comment (lightningterror) [inline bin/GameIndex.dbf]: Ditto.

**Pull request merged**